### PR TITLE
Set control-plane-ready annotation on Cluster

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:
This PR sets an annotation on the linked Machine's cluster object when the first control-plane instance is in a running state and has a NodeRef.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/controller/machine/machine_controller_phases.go [hunk 1]
```diff
@@ -251,3 +252,25 @@ func (r *ReconcileMachine) reconcileInfrastructure(ctx context.Context, m *v1alp
...
+// reconcileClusterAnnotations reconciles the annotations on the Cluster associated with Machines.
+// TODO(vincepri): Move to cluster controller once the proposal merges.
+func (r *ReconcileMachine) reconcileClusterAnnotations(ctx context.Context, cluster *v1alpha2.Cluster, m *v1alpha2.Machine) error {
+	if !m.DeletionTimestamp.IsZero() || cluster == nil {
+		return nil
+	}
+
+	// If the Machine is a control plane, it has a NodeRef and it's ready, set an annotation on the Cluster.
+	if util.IsControlPlaneMachine(m) && m.Status.NodeRef != nil && m.Status.InfrastructureReady {
+		if _, ok := cluster.Annotations[v1alpha2.ClusterAnnotationControlPlaneReady]; !ok {
```
Comment (ncdc) [inline pkg/controller/machine/machine_controller_phases.go]: We'll need to nil-check the Annotations map and create it if needed

Comment (vincepri) [reply pkg/controller/machine/machine_controller_phases.go]: Done, good catch!

File: pkg/apis/cluster/v1alpha2/zz_generated.deepcopy.go [hunk 2]
```diff
@@ -225,6 +225,42 @@ func (in *Machine) DeepCopyObject() runtime.Object {
 	return nil
 }
 
+// DeepCopyInto is an autogenerated deepcopy function, copying the receiver, writing into out. in must be non-nil.
```
Comment (detiber) [inline pkg/apis/cluster/v1alpha2/zz_generated.deepcopy.go]: Do we need a pull-related check to validate that these are up to date?

Comment (vincepri) [reply pkg/apis/cluster/v1alpha2/zz_generated.deepcopy.go]: Agreed, I'll open an issue

Comment (vincepri) [reply pkg/apis/cluster/v1alpha2/zz_generated.deepcopy.go]: Filed https://github.com/kubernetes-sigs/cluster-api/issues/1172

**Pull request merged**